File: dd demo/src/main/java/andrew/volostnykh/dddemo/model/DistributionContract.java
package andrew.volostnykh.dddemo.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import static andrew.volostnykh.dddemo.model.DistributionContract.DISTRIBUTION_CONTRACTS;

@Builder
@Data
@Entity
@Table(name = DISTRIBUTION_CONTRACTS)
@NoArgsConstructor
@AllArgsConstructor
public class DistributionContract implements Model {
    public static final String DISTRIBUTION_CONTRACTS = "distribution_contracts";

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private BigDecimal price;
    private BigDecimal delayFine;
    private LocalDateTime startDate;
    private LocalDateTime deadLine;
    private BigDecimal depreciationRate;
    private BigDecimal insuranceRate;

    @Override
    public String getEntityName() {
        return DISTRIBUTION_CONTRACTS;
    }
}

File: dd-demo-fe/src/app/model/distribution-contract.ts
import {Detail} from './detail';

export class DistributionContract {
  delayFine:number;
  depreciationRate:number;
  insuranceRate:number;
  price:number;
  deadLine:Date;
  startDate:Date;
  id:number;
  description:string;
  name:string;

  details: Detail[]

  constructor(id: number,
              name: string,
              delayFine: number,
              depreciationRate: number,
              insuranceRate: number,
              price: number,
              deadLine: Date,
              startDate: Date,
              description: string,
              details: Detail[]) {
    this.delayFine = delayFine;
    this.depreciationRate = depreciationRate;
    this.insuranceRate = insuranceRate;
    this.price = price;
    this.deadLine = deadLine;
    this.startDate = startDate;
    this.id = id;
    this.description = description;
    this.name = name;
    this.details = details;
  }
}
